pkg/provider/serviceprovider/serviceprovider.go

```diff
@@ -35,17 +33,7 @@ func (sp *ServiceProvider) LoginURL(id string) string {
 }
 
 func NewServiceProvider(id string, config *Config, defaultLoginURL string) (*ServiceProvider, error) {
-	metadataData := make([]byte, 0)
-	if config.URL != "" {
-		body, err := xml.ReadMetadataFromURL(config.URL)
-		if err != nil {
-			return nil, err
-		}
-		metadataData = body
-	} else {
-		metadataData = config.Metadata
-	}
-	metadata, err := xml.ParseMetadataXmlIntoStruct(metadataData)
+	metadata, err := xml.ParseMetadataXmlIntoStruct(config.Metadata)
```

Comment:
don't we want to support metadata via url as well?

Reply:
In my opinion as you still have the option to use the xml.ReadMetadataFromURL function, we still support it, but as we currently don't want to read the metadata again for each request so I removed the part in the NewServiceProvider.